gulpfile.js

```diff
@@ -39,80 +47,130 @@ gulp.task('lint', function() {
+    .pipe(
+      rollup({
+        sourceMap: true,
+        format: 'amd',
+        external: [
+          'components/mvc',
+          'components/react',
+          'components/react-dom',
+        ],
+        plugins: [
+          rollupBabelPlugin({ presets: ['es2015-rollup'] }),
+          rollupIncludePathsPlugin({ extensions: ['.js', '.jsx'] }),
+        ],
+      })
+    )
+    .pipe(sourcemaps.write('.'))
+    .pipe(gulp.dest(`${DIST_APP_ROOT}js`));
 });
 
 /**
- * Apply Babel transform to the JS and JSX files.
+ * Compiles app source for production i.e. processes source with Bable, Rollup
+ * and UglifyJS.
  */
-gulp.task('babel-app', function() {
-  const files = [
-    `${APP_ROOT}js/**/*.{js,jsx}`,
-  ];
-
-  try {
-    return gulp.src(files)
-      .pipe(babel().on('error', function(e) {
-          console.log('error running Babel', e);
-        })
-      )
-      .pipe(gulp.dest(`${DIST_APP_ROOT}js/`));
-  } catch (e) {
-    console.log('Got error in Babel', e);
-  }
+gulp.task('compile-app-production', ['compress-external-modules'], function() {
+  return gulp
+    .src(`${APP_ROOT}js/*.js`)
+    .pipe(
+      rollup({
+        sourceMap: true,
+        format: 'amd',
+        external: [
+          'components/mvc',
+          'components/react',
+          'components/react-dom',
+        ],
+        plugins: [
+          rollupBabelPlugin({ presets: ['es2015-rollup'] }),
+          rollupUglifyPlugin({ sourceMap: true }, uglifyjs.minifier),
+          rollupIncludePathsPlugin({ extensions: ['.js', '.jsx'] }),
+        ],
+      })
+    )
+    .pipe(sourcemaps.write('.'))
+    .pipe(gulp.dest(`${DIST_APP_ROOT}js`));
 });
 
 /**
- * Process unit test files with Babel.
+ * Compresses external modules that don't provide compressed version.
  */
-gulp.task('babel-unit-tests', function() {
-  try {
-    return gulp.src([`${TESTS_ROOT}unit/**/*.js`])
-      .pipe(
-        babel().on('error', function(e) {
-          console.error('Error occurred while running Babel', e);
-        })
-      )
-      .pipe(gulp.dest(`${DIST_TESTS_ROOT}unit/`));
-  } catch (e) {
-    console.error('Error occurred while process unit test files with Babel', e);
-  }
+gulp.task('compress-external-modules', function() {
+  return gulp
+    .src([
+      `${DIST_APP_ROOT}js/alameda.js`,
+      `${DIST_APP_ROOT}js/components/mvc.js`,
+      `${DIST_APP_ROOT}js/polyfills/fetch.js`,
+    ], { base: DIST_APP_ROOT })
+    .pipe(sourcemaps.init())
+    .pipe(minifier({}, uglifyjs))
+    .pipe(sourcemaps.write('.'))
+    .pipe(gulp.dest(DIST_APP_ROOT));
 });
 
-gulp.task('remove-useless', function() {
-  return del([
-    `${DIST_APP_ROOT}js/**/*.jsx`,
-    `${DIST_APP_ROOT}**/*.md`,
-  ]);
+/**
+ * Applies Babel transform to the app/tests JS and JSX files. The only
+ * difference between normal app build and build for unit tests is that Rollup
+ * is not used here.
+ */
+gulp.task('compile-unit-tests', function() {
+  return merge(
+    gulp
```

Comment:
Here is the thing I don't really like, but I've struggled with Rollup & Unit tests too much already :) Ideally I wanted to run unit tests on the production build, but rollup doesn't export anything from the bundle...